### ISSUE TITLE
Password reset page broken after user opens reset token

### ISSUE THREAD

**Issue by monsterpit-daggertooth** [glitch-soc/mastodon#360]
_(HEAD at commit e815d6.)_

When a user resets their password and clicks the reset token link, they receive an error 500.

In the Puma log:
```
method=GET path=/auth/password/edit format=html controller=Auth::PasswordsController action=edit status=401 duration=26.28 view=0.00 db=1.50

ActionView::Template::Error (undefined method `encrypted_password' for nil:NilClass):
    4:   = simple_form_for(resource, as: resource_name, url: password_path(resource_name), html: { method: :put }) do |f|
    5:     = render 'shared/error_messages', object: resource
    6:
    7:     - if use_pam? || current_user.encrypted_password.present?
    8:       = f.input :reset_password_token, as: :hidden
    9:
   10:       = f.input :password, autofocus: true, placeholder: t('simple_form.labels.defaults.new_password'), input_html: { 'aria-label' => t('simple_form.labels.defaults.new_password'), :autocomplete => 'off' }

app/views/auth/passwords/edit.html.haml:7:in `block (2 levels) in _app_views_auth_passwords_edit_html_haml___3409877845550787849_69990553306380'
app/views/auth/passwords/edit.html.haml:4:in `block in _app_views_auth_passwords_edit_html_haml___3409877845550787849_69990553306380'
app/views/auth/passwords/edit.html.haml:1:in `_app_views_auth_passwords_edit_html_haml___3409877845550787849_69990553306380'
```
* * * *

- [X] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by yipdw** [comment]
The error implies `current_user` is nil, which I _think_ ought to be an expected case (that the template clearly isn't expecting).  

tootsuite/mastodon contains much the same code in `Auth::PasswordsController` and `auth/passwords/edit`, so I think this might be a bug in tootsuite.  I'll try to confirm there (or if someone else beats me to it, that'd be great).

**Comment by ThibG** [comment]
It's a bug upstream that has been fixed in 411c9ecb4b7396afe95bbf6f191616dcd3fc970c

**Comment by ThibG** [comment]
Should have been fixed by merging upstream. Please reopen if that's not the case.
